Microsoft.Research/Contracts/WindowsBase/System.Windows.Rect.cs

```diff
@@ -339,7 +352,7 @@ public double Width
       }
       set
       {
-        Contract.Requires(!this.IsEmpty);
+        // Contract.Requires(!this.IsEmpty); => Is true, but impossible to proof with acceptable effort.
```

Comment:
:thought_balloon: It would be really cool if the contract block allowed the use of `Contract.Assume`. For example, this method could use the following contract:

```csharp
Contract.Assume(!this.IsEmpty); // Required, but not currently provable statically
Contract.Requires(value >= 0.0 || Double.IsNaN(value));
```